<a href="https://colab.research.google.com/github/Moulika-tata/AI-ML-Engineer-90-Days/blob/main/Day_02_Python_Operators_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Day 2 — Python Operators

## Topics Covered

1. Arithmetic Operators
2. Comparison Operators
3. Logical Operators
4. Assignment Operators
5. Operator Precedence
6. Practice Programs
7. Interview Questions
8. Final MCQ Test

## What is an Operator?

An operator is a symbol or keyword used to perform an operation on values or variables.

Example:

a = 10
b = 5

a + b

Here:
- a and b are operands
- + is the operator
- a + b is an expression

## 1. Arithmetic Operators

Arithmetic operators are used to perform mathematical calculations.

+   → Addition
-   → Subtraction
*   → Multiplication
/   → Division
//  → Floor Division
%   → Modulo / Remainder
**  → Exponentiation

Examples:

10 + 5 = 15
10 - 5 = 5
10 * 5 = 50
10 / 5 = 2.0
10 // 3 = 3
10 % 3 = 1
2 ** 3 = 8

In [8]:
a = 10
b = 3

print("Addition =", a + b)
print("Subtraction =", a - b)
print("Multiplication =", a * b)
print("Division =", a / b)
print("Floor Division =", a // b)
print("Remainder =", a % b)
print("Power =", a ** b)

Addition = 13
Subtraction = 7
Multiplication = 30
Division = 3.3333333333333335
Floor Division = 3
Remainder = 1
Power = 1000


## 2. Comparison Operators

Comparison operators are used to compare two values.

The result is always True or False.

>   → Greater than
<   → Less than
>=  → Greater than or equal to
<=  → Less than or equal to
==  → Equal to
!=  → Not equal to

Important:

=  → Assignment
== → Comparison

In [9]:
a = 10
b = 5

print(a > b)
print(a < b)
print(a >= b)
print(a <= b)
print(a == b)
print(a != b)

True
False
True
False
False
True


## 3. Logical Operators

Logical operators are used to combine or reverse conditions.

and → Both conditions must be True
or  → At least one condition must be True
not → Reverses the Boolean value

Examples:

True and True → True
True and False → False

True or False → True
False or False → False

not True → False
not False → True

In [10]:
print(True and True)
print(True and False)

print(True or False)
print(False or False)

print(not True)
print(not False)

True
False
True
False
False
True


## 4. Assignment Operators

Assignment operators are used to assign or update values in variables.

=    → Assignment
+=   → Add and assign
-=   → Subtract and assign
*=   → Multiply and assign
/=   → Divide and assign
//=  → Floor divide and assign
%=   → Modulo and assign
**=  → Power and assign

Examples:

x += 5
means
x = x + 5

x *= 2
means
x = x * 2

In [11]:
x = 10

x += 5
print("After += :", x)

x -= 3
print("After -= :", x)

x *= 2
print("After *= :", x)

x //= 2
print("After //= :", x)

x %= 3
print("After %= :", x)

After += : 15
After -= : 12
After *= : 24
After //= : 12
After %= : 0


## 5. Operator Precedence

Python follows an order when evaluating expressions.

Order:

1. Parentheses ()
2. Exponentiation **
3. Multiplication *, Division /, Floor Division //, Modulo %
4. Addition +, Subtraction -

Example:

2 + 3 * 4

First:
3 * 4 = 12

Then:
2 + 12 = 14

Therefore:
2 + 3 * 4 = 14

Parentheses can change the order:

(2 + 3) * 4 = 20

In [12]:
print(2 + 3 * 4)
print((2 + 3) * 4)
print(2 ** 3 + 1)
print(10 - 2 * 3)

14
20
9
4


# Practice Programs

Today I practiced the following programs:

1. Calculate Total and Average
2. Calculate Product and Remainder
3. Check Divisibility by 5
4. Check Positive, Negative, or Zero
5. Compare Two Numbers
6. Check Even and Divisible by 5
7. Check Eligibility
8. Calculate Discount

## Program 1 — Calculate Total and Average

Take three numbers from the user.

Calculate:
- Total
- Average

In [13]:
num1 = int(input("Enter first number: "))
num2 = int(input("Enter second number: "))
num3 = int(input("Enter third number: "))

total = num1 + num2 + num3
average = total / 3

print("Total =", total)
print("Average =", average)

Enter first number: 23
Enter second number: 34
Enter third number: 56
Total = 113
Average = 37.666666666666664


## Program 2 — Product and Remainder

Take two integers from the user.

Calculate:
- Product
- Remainder

In [14]:
num1 = int(input("Enter first number: "))
num2 = int(input("Enter second number: "))

product = num1 * num2
remainder = num1 % num2

print("Product =", product)
print("Remainder =", remainder)

Enter first number: 23
Enter second number: 34
Product = 782
Remainder = 23


## Program 3 — Divisibility by 5

Check whether a number is divisible by 5.

In [15]:
num = int(input("Enter number: "))

if num % 5 == 0:
    print("Divisible by 5")
else:
    print("Not divisible by 5")

Enter number: 4
Not divisible by 5


## Program 4 — Positive, Negative, or Zero

Check whether a number is:
- Positive
- Negative
- Zero

In [16]:
num = int(input("Enter number: "))

if num > 0:
    print("Positive")
elif num < 0:
    print("Negative")
else:
    print("Zero")

Enter number: 4
Positive


## Program 5 — Compare Two Numbers

Take two numbers and determine:
- First number is greater
- Second number is greater
- Both numbers are equal

In [17]:
num1 = int(input("Enter first number: "))
num2 = int(input("Enter second number: "))

if num1 > num2:
    print("First number is greater")
elif num1 == num2:
    print("Both numbers are equal")
else:
    print("Second number is greater")

Enter first number: 34
Enter second number: 56
Second number is greater


## Program 6 — Even and Divisible by 5

Check whether a number is both:
- Even
- Divisible by 5

Both conditions must be True.

In [18]:
num = int(input("Enter number: "))

if num % 2 == 0 and num % 5 == 0:
    print("Even and divisible by 5")
else:
    print("Condition not satisfied")

Enter number: 6
Condition not satisfied


## Program 7 — Eligibility Check

A person is eligible if:

Age >= 18
AND
Marks >= 60

In [19]:
age = int(input("Enter age: "))
marks = int(input("Enter marks: "))

if age >= 18 and marks >= 60:
    print("Eligible")
else:
    print("Not Eligible")

Enter age: 67
Enter marks: 78
Eligible


## Program 8 — Discount Calculator

If the price is 1000 or more:
- Give a 10% discount.

Otherwise:
- No discount.

In [20]:
price = int(input("Enter price: "))

if price >= 1000:
    discount = price * 10 / 100
    final_price = price - discount

    print("Discount =", discount)
    print("Final price =", final_price)
else:
    print("Discount = 0")
    print("Final price =", price)

Enter price: 2000
Discount = 200.0
Final price = 1800.0


## Interview Questions

### 1. What is an operator in Python?

An operator is a symbol or keyword used to perform an operation on values or variables.

### 2. What is the difference between /, //, and %?

/ performs normal division and returns a floating-point result.

// performs floor division and returns the floor value.

% returns the remainder of a division.

Example:

10 / 3  → 3.333...
10 // 3 → 3
10 % 3  → 1

### 3. What is the difference between = and ==?

= is used for assignment.

== is used to compare two values for equality.

# Day 2 Final Test

Final MCQ Score: 14/15
Percentage: 93.3%

Day 2 Status: Completed ✅

Topics completed:
- Arithmetic Operators
- Comparison Operators
- Logical Operators
- Assignment Operators
- Operator Precedence
- Python Practice Programs
- Interview Questions
- Final MCQ Test